### PR TITLE
Begin integrating support for themes

### PR DESCRIPTION
**Pull request by j-f1**
This is extracted from #922. In this PR:

- Feature flag for themes
- Theme selector in the titlebar & settings when the flag is enabled
- New color palette (used only for the theme/dynamic colors). ~QUESTION: should we use this palette or keep the existing one? (third option: keep the existing palette, but update the color names to match the new palette) [Compare them here](https://observablehq.com/d/1add1aaf4aee8d4e)~ (keeping existing palette for now)
- reorganize the `styles.tsx` file a bit to make it easier to understand.
- CSS variables are dynamically injected into a `<style>` tag to allow the new dynamic colors to work
  - This happens even if the feature flag is disabled. As we port components to use dynamic colors, they will depend on these CSS variables, I think it makes sense to just force the theme to light mode rather than trying to keep the old coloring somehow as we port components over.
- Added the feature flag to the feature flag settings, rewrote the feature flag section in TS to make it easier to expand on for future flags

Not included in this PR:

- any visual changes besides the new UI components

### CODE REVIEW

File: packages/desktop-client/src/style/colors.ts
Comment (j-f1) [inline packages/desktop-client/src/style/colors.ts:1]: This is unchanged from `master`, just reformatted as its own file.

File: packages/desktop-client/src/style/themes/dark.ts
Comment (j-f1) [inline packages/desktop-client/src/style/themes/dark.ts:1]: Pulled directly from #922 and reformatted.

File: packages/desktop-client/src/components/settings/Themes.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import { Setting } from './UI';
+
+export default function ThemeSettings({ saveGlobalPrefs }) {
+  let theme = useTheme();
+  return (
+    <Setting
+      primaryAction={
+        <View>
+          <Select
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/settings/Themes.js:14]: **💬 suggestion:** ‏missing background color

Screenshot 2023-07-20 at 08 25 45

File: packages/desktop-client/src/components/ThemeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+export function ThemeSelector() {
+  let theme = useTheme();
+  let { saveGlobalPrefs } = useActions();
+
+  let { isNarrowWidth } = useResponsive();
+
+  // Don't display on mobile
+  return isNarrowWidth ? null : (
+    <View>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/ThemeSelector.tsx:19]: **🥜 nitpick:** ‏do we need the wrapping `View`?

**Review by MatissJanis — COMMENTED** [review]
Currently clicking the theme switch button does not change any colors. That's expected, right?

**Comment by j-f1** [comment]
That’s expected, and it finally fulfills the promise of an experimental feature doing nothing at all ;)

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**